test/util/e2e.go

```diff
@@ -73,19 +73,22 @@ func waitForOperatorAvailability(ctx context.Context, k8sClient client.Client, k
 	pods := &corev1.PodList{}
 	gomega.EventuallyWithOffset(2, func(g gomega.Gomega) error {
 		g.Expect(k8sClient.Get(ctx, key, deployment)).To(gomega.Succeed())
-		g.Expect(deployment.Status.Conditions).To(gomega.ContainElement(gomega.BeComparableTo(
```

Comment:
actually, it makes sense to check the condition first. Otherwise the pods might not have been created yet.

Reply:
My concern here is to don't wait 5 minutes if it restarts. Just stop at the first detection. If it restarts, the deployment will be unavailable and we won't be able to test it. If it is not created, it will just be an empty array, so this case will be skipped and we will move on to validating the deployments.

Reply:
I guess the likelihood that there are no pods and the deployment reports ready in the same iteration are minimal.

ok